### PR TITLE
[Feature](bangc-ops): three_interpolate supports the tensor num near …

### PR DESCRIPTION
**Pull request by ArtIntAI**
…to 2g.

Thanks for your contribution and we appreciate it a lot. 

## 1. Motivation

To support the near to 2g number cases for three_interpolate_forward and three_interpolate_backward operators.

## 2. Modification

ThreeInterpolate

## 3. Test Report

If you want to know how to do operator testing, you can see [GTest-User-Guide-zh](https://github.com/Cambricon/mlu-ops/blob/master/docs/GTest-User-Guide-zh.md).

#### 3.1.1 Accuracy Acceptance Standard

For static threshold standard details, see: [MLU-OPS Accuracy Acceptance Standard](https://github.com/Cambricon/mlu-ops/blob/master/docs/MLU-OPS-Accuracy-Acceptance-Standard.md).

- [ ] diff1: diff1 <= 3e-3
- [ ] diff2: diff2 <= 3e-3

#### 3.1.2 Operator Scheme checklist

|     No.        |                 Details              |            Check Results             |
|----------------|--------------------------------------|--------------------------------------|
|        1       |Supported hardware                    |             MLU370<br>MLU590         |
|        2       |Job types                             |          block <br> U1 <br> U4       |
|        3       |Layouts                               |          NHWC 、NCHW、ARRAY etc      |
|        4       |Whether multi-dimensions are supported|                                      |
|        5       |Whether element zero is supported     |                                      |
|        6       |Data type(half/float)                 |           half / float etc           |
|        7       |Whether there is size limit           |                                      |

#### 3.1.3 New Feature Test

If you have checked the following items, please tick the relevant box.

- [ ] Data type test
- [ ] Multi-dimensional tensor test
- [ ] Layout test
- [ ] Different size/integer remainder end segment/alignment misalignment test
- [ ] Zero dimensional tensor test/zero element test
- [ ] stability test
- [ ] Multiple platform test
- [ ] Gen_case module test
- [ ] Nan/INF tests 
- [ ] Bug fix tests
- [ ] For memory leak check details, see[GTest-User-Guide-zh](https://github.com/Cambricon/mlu-ops/blob/master/docs/GTest-User-Guide-zh.md).
- [ ] For code coverage check details, see: [GTest-User-Guide-zh](https://github.com/Cambricon/mlu-ops/blob/master/docs/GTest-User-Guide-zh.md).
- [ ] For I/O calculation efficiency check details, see: [MLU-OPS Performance Acceptance Standard](https://github.com/Cambricon/mlu-ops/blob/master/docs/MLU-OPS-Performance-Acceptance-Standard.md).

#### 3.1.4 Parameter Check

When a new operator is submitted, the test points are given and the test results are stated.

|                   Test Point                    | Acceptance Standard | Test Result (Error Message) |
| ----------------------------------------------- | --------------------| --------------------------- |
| Whether it conforms to the operator restriction |     Normal error    |                             |
| Whether illegal parameters are passed           |     Normal error    |                             |

### 3.2 Accuracy Test

For the cases used in the New Feature Test section, the features and the number of cases are recorded here. When multiple operations are tested, multiple tables are needed to include details of these operations.

Operation:

|Test Point           | Description                      | Quantity |  Comment |
|----------           |----------------------------------|----------|  --------|
|Data type test       |half/float/int8                   |          |          |
|Mult-tensor test     |Supports 1-8 dims                 |          |          |
|Layout test          |Supports NCHW/NHWC                |          |          |
|Zero element test    |Whether to support this test      |          |          |
|Stability test       |--gtest_repeat=NUM<br>--thread=NUM|          |          |
|Mult-platform test   |MLU370/MLU590                     |          |          |
|Nan/INF test         |Whether to support this test      |          |          |
|Memory leak check    |Test result                       |          |          |
|Code coverage check  |Test result                       |          |          |

回归测试结果：
[----------] Global test environment tear-down
[ SUMMARY  ] Total 560 cases of 2 op(s).
ALL PASSED.
[==========] 560 test cases from 2 test suites ran. (396523 ms total)
[  PASSED  ] 560 test cases.

接近2G num的新增case测试结果：
[----------] Global test environment tear-down
[ SUMMARY  ] Total 12 cases of 2 op(s).
ALL PASSED.
[==========] 12 test cases from 2 test suites ran. (272699 ms total)
[  PASSED  ] 12 test cases.

### 3.3 Performance Test

See [MLU-OPS Performance Acceptance Standard](https://github.com/Cambricon/mlu-ops/blob/master/docs/MLU-OPS-Performance-Acceptance-Standard.md) for details.

Platform：MLU370

|Operation|Mlu_hardware_time(us)|Mlu_interface_time(us)|Mlu_io_efficiency|Mlu_compute_efficiency|Mlu_workwpace_size(Bytes)|Data_type|Shape|
|-------|----|----|-----|----|----|----|-----|
|op_name|    |    |     |    |    |    |     |
|op_name|    |    |     |    |    |    |     |

Platform：MLU590

|Operation|Mlu_hardware_time(us)|Mlu_interface_time(us)|Mlu_io_efficiency|Mlu_compute_efficiency|Mlu_workwpace_size(Bytes)|Data_type|Shape|
|-------|----|----|----|----|----|----|-----|
|op_name|    |    |    |    |    |    |     |
|op_name|    |    |    |    |    |    |     |

### 3.4 Summary Analysis

Please give a brief overview here, if you want to note and summarize the content.

### CODE REVIEW

File: bangc-ops/kernels/three_interpolate/three_interpolate_union1.mlu [hunk 1]
```diff
@@ -35,20 +35,106 @@ __nram__ char nram_buffer[MAX_NRAM_SIZE];
 #define INT16_MAX_MASK 0xffff
 #define INT32_MASK_REPEAT_TIMES 4
 #define INT16_MASK_REPEAT_TIMES 2
+#define INT32_MAX_NUM 2147483647
+
+template <typename T>
+__mlu_func__ void G2N2DCopy(T *dst, const T *src, uint32_t data_size,
```
Comment (DanieeelLiu) [inline bangc-ops/kernels/three_interpolate/three_interpolate_union1.mlu:41]: 建议G2N2DCopy 名字修改下，改为load store 等更合适

File: bangc-ops/kernels/three_interpolate/three_interpolate_union1.mlu [hunk 2]
```diff
@@ -464,12 +571,22 @@ __mlu_global__ void MLUKernelThreeInterpolateBackward(
                  base_addr_weights + k * n_limit * INDEX_WEIGHT_LAST_DIM,
                  real_indices_deal_size * sizeof(T), GDRAM2NRAM);
         // load grad_output
-        __memcpy(nram_grad_output,
-                 base_addr_grad_output + (j * n * c_limit + k * n_limit),
-                 n_slice * sizeof(T), GDRAM2NRAM, n_limit_new * sizeof(T),
-                 n * sizeof(T), c_slice - 1);
+        uint32_t dst_stride = n_limit_new * sizeof(T);
+        uint32_t src_stride = n * sizeof(T);
```
Comment (BetterDongW) [inline bangc-ops/kernels/three_interpolate/three_interpolate_union1.mlu:575]: 当T是float时，可能溢出；建议修改成unit64

**Review by BetterDongW — APPROVED** [review]

**Review by GuoZhiBin2014 — APPROVED** [review]

File: bangc-ops/kernels/three_interpolate/three_interpolate_union1.mlu (hunk 2 above)
Comment (ArtIntAI) [reply bangc-ops/kernels/three_interpolate/three_interpolate_union1.mlu]: 已经修改。

File: bangc-ops/kernels/three_interpolate/three_interpolate_union1.mlu (hunk 1 above)
Comment (ArtIntAI) [reply bangc-ops/kernels/three_interpolate/three_interpolate_union1.mlu]: 已经修改。

**Review by DanieeelLiu — APPROVED** [review]

**Pull request merged**